DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -30,20 +40,41 @@ namespace edmNew {
 
   namespace dslv {
     template< typename T> class LazyGetter;
+
   }
 
   /* transient component of DetSetVector
-   * for pure conviniency of dictioanary declaration
+   * for pure conviniency of dictionary declaration
    */
   namespace dstvdetails {
+
+    void errorFilling();
+    void notSafe();
+    void errorIdExists(det_id_type iid);
+    void throw_range(det_id_type iid);
+
     struct DetSetVectorTrans {
       DetSetVectorTrans(): filling(false){}
-      bool filling;
+#ifndef USE_ATOMIC
```

Comment:
I believe non-persistent ```std::atomic```s don't appear to cause ROOT 6 any problems. Therefore ```filling``` shouldn't need to be protected.